Notation.Plugin.AzureKeyVault/KeyVault/KeyVaultClient.cs

```diff
@@ -43,50 +43,47 @@ private record KeyVaultMetadata(string KeyVaultUrl, string Name, string Version)
         protected Lazy<CryptographyClient> _cryptoClient;
         // Secret client (lazy initialization)
         protected Lazy<SecretClient> _secretClient;
-        // Error message for invalid input
-        private const string INVALID_INPUT_ERROR_MSG = "Invalid input. The valid input format is '{\"contractVersion\":\"1.0\",\"keyId\":\"https://<vaultname>.vault.azure.net/<keys|certificate>/<name>/<version>\"}'";
 
         // Key name or certificate name
         private string _name;
         // Key version or certificate version
-        private string _version;
+        private string? _version;
         // Key identifier (e.g. https://<vaultname>.vault.azure.net/keys/<name>/<version>)
         private string _keyId;
 
         // Internal getters for unit test
         internal string Name => _name;
-        internal string Version => _version;
+        internal string? Version => _version;
         internal string KeyId => _keyId;
 
         /// <summary>
         /// Constructor to create AzureKeyVault object from keyVaultUrl, name 
         /// and version.
         /// </summary>
-        public KeyVaultClient(string keyVaultUrl, string name, string version, TokenCredential credential)
+        public KeyVaultClient(string keyVaultUrl, string name, string? version, TokenCredential credential)
         {
             if (string.IsNullOrEmpty(keyVaultUrl))
             {
-                throw new ArgumentNullException(nameof(keyVaultUrl), "KeyVaultUrl must not be null or empty");
+                throw new ValidationException("Key vault URL must not be null or empty");
             }
 
             if (string.IsNullOrEmpty(name))
             {
-                throw new ArgumentNullException(nameof(name), "KeyName must not be null or empty");
+                throw new ValidationException("Key name must not be null or empty");
             }
 
-            if (string.IsNullOrEmpty(version))
+            _name = name;
+            _version = version;
+            _keyId = $"{keyVaultUrl}/keys/{name}";
+            if (!string.IsNullOrEmpty(version))
```

Comment:
`version` can be `null` but cannot be empty.

Reply:
Updated to validation to ensure `version` is not empty.